### PR TITLE
Allow new major versions of `bundler` and `public_suffix` dependencies

### PR DESCRIPTION
**Pull request by AlexWayfer**

### CODE REVIEW

File: rack-protection-maximum_cookie.gemspec [hunk 1]
```diff
@@ -22,10 +22,10 @@ Gem::Specification.new do |spec|
 
   spec.required_ruby_version = '>= 2.1.10'
 
-  spec.add_dependency 'public_suffix', '~> 3.0'
+  spec.add_dependency 'public_suffix', '~> 4.0'
```
Comment (mwpastore) [inline rack-protection-maximum_cookie.gemspec]: I'm thinking `'>= 3.0', '< 5'` instead.

File: rack-protection-maximum_cookie.gemspec [hunk 2]
```diff
@@ -22,10 +22,10 @@ Gem::Specification.new do |spec|
...
   spec.add_dependency 'rack', ENV.fetch('RACK_VERSION', ['>= 1.4.7', '< 2.1'])
 
-  spec.add_development_dependency 'bundler', '~> 1.15'
+  spec.add_development_dependency 'bundler', '~> 2.0'
```
Comment (mwpastore) [inline rack-protection-maximum_cookie.gemspec]: I'm thinking `'>= 1.15', '< 3'` instead.

**Comment by mwpastore** [comment]
Hi @AlexWayfer, thank you for the PR! My initial reaction is that we don't want to break compatibility with Bundler 1.x and older Rubies (public_suffix 4.0+ requires Ruby 2.3+). Take a look at my notes on your changes and let me know your thoughts.

**Comment by AlexWayfer** [comment]
> My initial reaction is that we don't want to break compatibility with Bundler 1.x and older Rubies (public_suffix 4.0+ requires Ruby 2.3+). Take a look at my notes on your changes and let me know your thoughts.

I thought about this approach, but I've decided to force new versions for this gem users. But, if you want this, I'll do.

**Comment by AlexWayfer** [comment]
Done.

**Comment by mwpastore** [comment]
Thank you!

**Pull request merged**